src/main/java/org/takes/rq/RqMultipart.java

```diff
@@ -506,6 +491,63 @@ private static StringBuilder fakeBody(final Request... dispositions)
                 .append(String.format("--%s--", BOUNDARY));
             return builder;
         }
+
+        /**
+         * This class is  for the specific implementation
+         * for the RqMultipart, (see the overridden methods for details).
+         *
+         * @todo #517:30min re-design the class.
+         *  Why the super class is encapsulated inside the class?
+         *  Find an implementation which won't be using
+         *  both composition and inheritance of the same class.
```

Comment:
@sebing I don't see anything bad in composition and inheritance of the same class, it's just decorator design pattern